lib/config.js

```diff
@@ -48,31 +48,39 @@ function putNestedValue(params, key, value) {
 function parseCloudinaryConfigFromEnvURL(ENV_STR) {
   let conf = {};
 
-  let uri = url.parse(ENV_STR, true);
+  const uri = new URL(ENV_STR);
+
+  const auth = uri.username && uri.password
+    ? `${uri.username}:${uri.password}`
+    : (uri.username || null);
 
   if (uri.protocol === 'cloudinary:') {
     conf = Object.assign({}, conf, {
-      cloud_name: uri.host,
-      api_key: uri.auth && uri.auth.split(":")[0],
-      api_secret: uri.auth && uri.auth.split(":")[1],
-      private_cdn: uri.pathname != null,
-      secure_distribution: uri.pathname && uri.pathname.substring(1)
+      cloud_name: uri.hostname,
+      api_key: uri.username || (auth && auth.split(":")[0]),
+      api_secret: uri.password || (auth && auth.split(":")[1]),
+      private_cdn: uri.pathname != null && uri.pathname !== '' && uri.pathname !== '/',
+      secure_distribution: uri.pathname && uri.pathname !== '/' ? uri.pathname.substring(1) : undefined
     });
   } else if (uri.protocol === 'account:') {
     conf = Object.assign({}, conf, {
-      account_id: uri.host,
-      provisioning_api_key: uri.auth && uri.auth.split(":")[0],
-      provisioning_api_secret: uri.auth && uri.auth.split(":")[1]
+      account_id: uri.hostname,
+      provisioning_api_key: uri.username || (auth && auth.split(":")[0]),
+      provisioning_api_secret: uri.password || (auth && auth.split(":")[1])
     });
   }
 
   return conf;
 }
 
 function extendCloudinaryConfigFromQuery(ENV_URL, confToExtend = {}) {
-  let uri = url.parse(ENV_URL, true);
-  if (uri.query != null) {
-    entries(uri.query).forEach(([key, value]) => putNestedValue(confToExtend, key, value));
+  const uri = new URL(ENV_URL);
+  if (uri.search) {
+    const query = {};
+    uri.searchParams.forEach((value, key) => {
```

Comment:
This could be done with `Object.fromEntries(uri.searchParams)` and I would rename `uri` to `url`.